hayeon/programmers-2/1차-캐시.js

```diff
@@ -0,0 +1,27 @@
+function solution(cacheSize, cities) {
+  const cache = [];
+  let answer = 0;
+
+  for (let i = 0; i < cities.length; i++) {
+    cities[i] = cities[i].toLowerCase();
+
+    const idx = cache.indexOf(cities[i]);
+    if (idx === -1) {
+      answer += 5;
+    } else {
+      answer += 1;
+      cache.splice(idx, 1);
+    }
+    cache.push(cities[i]);
+    if (cache.length > cacheSize) cache.shift();
+  }
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(cacheSize, cities) {
  const cache = [];
  let answer = 0;

  for (let i = 0; i < cities.length; i++) {
    const loweredCity = cities[i].toLowerCase();
    const idx = cache.indexOf(loweredCity);
    
    if (idx === -1) {
      answer += 5;
    } else {
      answer += 1;
      cache.splice(idx, 1);
    }
    cache.push(loweredCity);
    if (cache.length > cacheSize) cache.shift();
  }

  return answer;
}
```
- 풀이 고생하셨습니다~ 간단한 loweredCity 변수하나만 추가해봤습니다